domain-ee/ee-ep-merge-app/src/python_src/graph_export.py

```diff
@@ -0,0 +1,139 @@
+            fixedsize="true",
+            width=0.2,
+            height=0.2,
+        )
+        node.set_fillcolor("black")
+        return node
+
+    def _initial_edge(self):
+        return pydot.Edge(
+            "i",
+            self.machine.initial_state.id,
+            label="",
+            color=SUCCESS,
+            fontname=self.font_name,
+            fontsize=self.transition_font_size,
+        )
+
+    def _state_as_node(self, state):
+        if 'failed' in state.name or 'failed' in state.name or 'error' in state.name:
+            color = ERROR
+        else:
+            color = SUCCESS
+
+        node = pydot.Node(
+            state.id,
+            label=f"{state.name}",
+            shape="rectangle",
+            style="rounded, filled",
+            fontname=self.font_name,
+            fontsize=self.state_font_size,
+            peripheries=2 if state.final else 1,
+            fillcolor="white",
+            color=color,
+        )
+
+        return node
+
+    def _transition_as_edge(self, transition):
+        if self.machine.main_event not in transition.events:
+            return None
+
+        if 'failed' in transition.source.name or 'failed' in transition.target.name or 'error' in transition.target.name:
+            color = ERROR
+        else:
+            color = SUCCESS
+
+        return pydot.Edge(
+            transition.source.id,
+            transition.target.id,
+            color=color,
+            fontname=self.font_name,
+            fontsize=self.transition_font_size,
+        )
+
+    def get_graph(self):
+        graph = self._get_graph()
+        graph.add_node(self._initial_node())
+        graph.add_edge(self._initial_edge())
+
+        for state in self.machine.states:
+            node = self._state_as_node(state)
+            if node:
+                if 'Completed success' == state.name:
+                    graph.add_node(node)
+                elif 'Completed error' == state.name:
+                    graph.add_node(node)
+                else:
+                    edges = []
+                    for transition in state.transitions:
+                        if transition.internal:
+                            continue
+                        edge = self._transition_as_edge(transition)
+                        if edge:
+                            edges.append(edge)
+                    if edges:
+                        graph.add_node(node)
+                        for edge in edges:
+                            graph.add_edge(edge)
+        return graph
+
+    def __call__(self):
+        return self.get_graph()
+
+
+def generate_graph(main_event):
+    job = MergeJob(job_id=uuid4(),
+                   pending_claim_id=1,
+                   ep400_claim_id=2)
+
+    #
+    graph = DotGraphMachine(EpMergeMachine(job, main_event))
+    dot = graph()
+    dot.write_png(f"EP_{main_event.replace('processing_from_running_','')}.png")
+
+
+if __name__ == "__main__":
```

Comment:
How does this module get used? I don't see it being called from anywhere else in this PR, so would you jump into a terminal and call it from the command line?

Reply:
Yeah its just an additional tool that generates those graphs that I posted in this slack [thread](https://dsva.slack.com/archives/C01BVF2A3V0/p1704475173178239).